packages/design-system/src/components/Dropdown/Dropdown.tsx

```diff
@@ -1,73 +1,101 @@
+import DropdownShell from './Primitive/DropdownShell';
+import DropdownTitle from './Primitive/DropdownTitle';
+import DropdownDivider from './Primitive/DropdownDivider';
 import Clickable, { ClickableProps } from '../Clickable';
+import { LinkableType } from '../Linkable';
 
-import * as S from './Dropdown.style';
+type DropdownButtonType = Omit<ClickableProps, 'children' | 'as'> & {
+	label: string;
+	onClick: () => void;
+	icon?: IconName;
+	type: 'button';
+};
+
+type DropdownLinkType = Omit<LinkableType, 'children'> & {
+	label: string;
+	type: 'link';
+};
 
-type DividerType = { divider: boolean };
-type LinkType = Omit<LinkableType, 'children'> & {
+type DropdownLabelType = {
+	type: 'title';
 	label: string;
 };
-type ButtonType = Omit<LinkType, 'href'> & Omit<ClickableProps, 'children'>;
-type MenuItemProps = DividerType | LinkType | ButtonType;
 
-export type DropdownProps = BoxProps &
-	StyledProps<any> & {
-		/** Dropdown menu items */
-		items: MenuItemProps[];
-	};
+type DropdownDividerType = {
+	type: 'divider';
+};
 
-function convertItem(item: ButtonType | LinkType) {
-	return 'href' in item ? (
-		<Linkable icon={item.icon} {...item}>
-			{item.label}
-		</Linkable>
-	) : (
-		<Clickable {...(item as ButtonType)}>{item.label}</Clickable>
-	);
-}
+type DropdownPropsType = {
+	children: ReactElement<typeof Clickable>;
+	items: (DropdownButtonType | DropdownLinkType | DropdownLabelType | DropdownDividerType)[];
+	'aria-label': string;
+};
 
-const Dropdown: React.FC<DropdownProps> = React.forwardRef(
-	({ children, items = [], ...props }: DropdownProps, ref) => {
+const Dropdown = forwardRef(
+	({ children, items, ...rest }: DropdownPropsType, ref: Ref<HTMLDivElement>) => {
 		const menu = useMenuState({
 			animated: 250,
-			gutter: 0,
+			gutter: 4,
 			loop: true,
 		});
-		const { 'aria-label': ariaLabel, 'aria-labelledby': ariaLabelledby, ...rest } = props;
 
 		return (
 			<>
-				<S.Button as={Clickable} data-test="dropdown.button" {...menu} {...rest} ref={ref}>
-					{children}
-					{items.length ? <S.ButtonIcon name="talend-caret-down" /> : null}
-				</S.Button>
-				{items.length ? (
-					<S.Menu {...menu} aria-label={ariaLabel} aria-labelledby={ariaLabelledby}>
-						<S.AnimatedMenu data-test="dropdown.menu" {...menu}>
-							{items.map((item: MenuItemProps, index: number) => {
-								if ('divider' in item) {
-									return <S.MenuSeparator key={`separator-${index}`} />;
-								}
+				<MenuButton {...menu} data-test="dropdown.button">
+					{disclosureProps => cloneElement(children, disclosureProps)}
+				</MenuButton>
+				<Menu {...menu} as={DropdownShell} {...rest} ref={ref} data-test="dropdown.menu">
+					{items.map((entry, index) => {
+						if (entry.type === 'button') {
+							const { label, ...entryRest } = entry;
+							return (
+								<DropdownButton
+									{...entryRest}
+									{...menu}
+									key={`${label}-${index}`}
+									id={`${label}-${index}`}
```

Comment:
Is it safe to have ids made of label + index ?
(Risk of duplicate ids in the same page)

Reply:
Nothing is ever safe. I don't know how to do better :/ I could remove the id, but Reakit prefers having it for perfs.

Reply:
React 18 `useId()` if not provided by user :D 
Let's go with that and see if we have an issue in the future